src/Elliptic/SubdomainPreconditioners/MinusLaplacian.hpp

```diff
@@ -0,0 +1,261 @@
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// Linear solvers that approximately invert the
+/// `elliptic::dg::subdomain_operator::SubdomainOperator` to make the Schwarz
+/// subdomain solver converge faster.
+/// \see LinearSolver::Schwarz::Schwarz
+namespace elliptic::subdomain_preconditioners {
+
+/// \cond
+template <size_t Dim, typename OptionsGroup, typename Solver,
+          typename LinearSolverRegistrars>
+struct MinusLaplacian;
+/// \endcond
+
+namespace Registrars {
+template <size_t Dim, typename OptionsGroup,
+          typename Solver = LinearSolver::Serial::LinearSolver<tmpl::list<
+              ::LinearSolver::Serial::Registrars::Gmres<
+                  ::LinearSolver::Schwarz::ElementCenteredSubdomainData<
+                      Dim, tmpl::list<Poisson::Tags::Field>>>,
+              ::LinearSolver::Serial::Registrars::ExplicitInverse>>>
+struct MinusLaplacian {
+  template <typename LinearSolverRegistrars>
+  using f = subdomain_preconditioners::MinusLaplacian<Dim, OptionsGroup, Solver,
+                                                      LinearSolverRegistrars>;
+};
+}  // namespace Registrars
+
+/*!
+ * \brief Approximate the subdomain operator with a flat-space Laplacian with
+ * Dirichlet boundary conditions for every tensor component separately.
+ *
+ * This linear solver applies the `Solver` to every tensor component in
+ * turn, approximating the subdomain operator with a flat-space Laplacian with
+ * Dirichlet boundary conditions. This can be a lot cheaper than solving the
+ * full subdomain operator and can provide effective preconditioning for an
+ * iterative subdomain solver. The approximation is better the closer the
+ * original PDEs are to a set of decoupled flat-space Poisson equations with
+ * Dirichlet boundary conditions.
+ *
+ * \tparam Dim Spatial dimension
+ * \tparam OptionsGroup The options group identifying the
+ * `LinearSolver::Schwarz::Schwarz` solver that defines the subdomain geometry.
+ * \tparam Solver Any class that provides a `solve` and a `reset` function,
+ * but typically a `LinearSolver::Serial::LinearSolver`. The solver will be
+ * factory-created from input-file options.
+ */
+template <size_t Dim, typename OptionsGroup,
+          typename Solver = LinearSolver::Serial::LinearSolver<tmpl::list<
+              ::LinearSolver::Serial::Registrars::Gmres<
+                  ::LinearSolver::Schwarz::ElementCenteredSubdomainData<
+                      Dim, tmpl::list<Poisson::Tags::Field>>>,
+              ::LinearSolver::Serial::Registrars::ExplicitInverse>>,
+          typename LinearSolverRegistrars =
+              tmpl::list<Registrars::MinusLaplacian<Dim, OptionsGroup, Solver>>>
+class MinusLaplacian
+    : public LinearSolver::Serial::LinearSolver<LinearSolverRegistrars> {
+ private:
+  using Base = LinearSolver::Serial::LinearSolver<LinearSolverRegistrars>;
+  using StoredSolverType = tmpl::conditional_t<std::is_abstract_v<Solver>,
```

Comment:
would it be possible to just always hold a pointer to simplify things?

Reply:
I tried that when building this class initially. The distinction is necessary for option-parsing, because `unique_ptr`s always dispatch to factory-creation, so at least the option tag and the constructor have to implement the pointer/value logic. At that point I found it easier to `std::move` the option-created object directly into the member variable and distinguish pointer/value through accessors. This also has the (very slight) advantage that pointer-dereferencing is avoided when it's not necessary.